### ISSUE TITLE
Rackspace snapshots - Force = true ignored

### ISSUE THREAD

**Issue by yannh** [fog/fog#3541]
So this call:

snapshot = volume.create_snapshot :display_name => snapshot_name, :force => true, :volume_id => volume.id

Fails with the following stack:

        from /var/lib/gems/1.9.1/gems/excon-0.45.3/lib/excon/middlewares/response_parser.rb:8:in `response_call'
        from /var/lib/gems/1.9.1/gems/excon-0.45.3/lib/excon/connection.rb:373:in `response'
        from /var/lib/gems/1.9.1/gems/excon-0.45.3/lib/excon/connection.rb:237:in `request'
        from /var/lib/gems/1.9.1/gems/fog-core-1.30.0/lib/fog/core/connection.rb:81:in `request'
        from /var/lib/gems/1.9.1/gems/fog-1.29.0/lib/fog/rackspace/service.rb:42:in `request'
        from /var/lib/gems/1.9.1/gems/fog-1.29.0/lib/fog/rackspace/block_storage.rb:89:in `request'
        from /var/lib/gems/1.9.1/gems/fog-1.29.0/lib/fog/rackspace/requests/block_storage/create_snapshot.rb:43:in `create_snapshot'
        from /var/lib/gems/1.9.1/gems/fog-1.29.0/lib/fog/rackspace/models/block_storage/snapshot.rb:65:in `save'
        from /var/lib/gems/1.9.1/gems/fog-core-1.30.0/lib/fog/core/collection.rb:51:in `create'
        from /var/lib/gems/1.9.1/gems/fog-1.29.0/lib/fog/rackspace/models/block_storage/volume.rb:91:in `create_snapshot'
        from ./snapshot.rb:65:in `<main>'

If we go up that trace:

volume.create_snapshot receives the options hash with :force set to true. It calls create in /lib/fog/core/collection.rb passing it the options - so in create, attributes[:force] is set to true.

Create then calls object.save in rackspace/models/block_storage/snapshot.rb without passing it any argument. Except that this save function, there is a parameter "force" that defaults to false. 

Long story short: it's not currently possible to take a snapshot of a disk that is connected to a VM using fog. 

I am trying to think of a fix but it won't be easy to fix without breaking backward compatibility...

**Comment by yannh** [comment]
Considering the fact that core/collection.rb calls save without parameters, and touching core is probably not what we want to do, adding an attribute :force to the Snapshot class seems inevitable to me.

Adding a :force attribute to the snapshot class and 

  force = self.force if self.force

at the beginning of the save function in the snapshot class fixes the issue while staying backward compatible. I hope someone finds a better solution though...

**Comment by geemus** [comment]
@yannh hey, sorry to hear about the difficulty. Adding the attribute to the class sounds pretty reasonable actually, why do you think that solution wouldn't be good?

**Comment by yannh** [comment]
Hi geemus, thanks for your reply! Well "force" isn't really a feature of a snapshot per se.. Do you think I should make a PR with the patch I proposed? I think it's really quite ugly..

**Comment by geemus** [comment]
Ah, ok, I think I see now. So you are saying that force should JUST be an argument to the save, as it is an option while saving, rather than an ongoing property of the object. (ie when you retrieve the object later it would never have a force attribute). Does that sound accurate?

If so, it seems like having something like `def save(options = {})` might be more appropriate? What do you think?

**Comment by yannh** [comment]
Yep, eventually.. But save is called from here: https://github.com/fog/fog-core/blob/master/lib/fog/core/collection.rb#L51  without arguments, and passing attributes to save() might need to change a lot of other code.. I admit I'm not familiar enough with fog's code to know what the nicest solution would be here.

**Comment by geemus** [comment]
I think we can probably just override that method in this case and change it a little as needed and/or super. I wouldn't be too concerned up front, at the very least I don't think it should hurt anything to give it a shot and see where it gets us. What do you think?

**Comment by yannh** [comment]
As much as I'd like to help, I'm not sure what you mean... Happy to test any patch though..

**Comment by geemus** [comment]
Sure, just may be a few days before I can get to it.

**Comment by geemus** [comment]
@mdarby Any chance you could take a look at this? Thanks!

**Comment by mdarby** [comment]
I'll take a look at it first thing tomorrow.

**Comment by geemus** [comment]
Awesome, thanks. It didn't look too crazy, but I also didn't dig in too
deep. Just let me know if you have questions or there is anything I can
help with.

**Comment by mdarby** [comment]
Working on this today -- I hope to have a PR EOD. Very sorry for the delay!

**Comment by mdarby** [comment]
I've submitted the above PR addressing this. I just added `force` as an attribute -- the [docs](http://developer.openstack.org/api-ref-blockstorage-v1.html#createSnapshot) show it in the payload and it not appearing in the response is kind of expected.

**Comment by geemus** [comment]
Thanks!

**Comment by yannh** [comment]
Hi mdarby, many thanks for the PR! Unfortunately I've changed company and can not test this.

**Comment by mdarby** [comment]
Still a bug! Thanks for finding and reporting it :)
